jlm/rvsdg/lambda.hpp

```diff
@@ -287,6 +314,81 @@ LambdaNode::RemoveLambdaInputsWhere(const F & match)
   return numRemovedInputs;
 }
 
+class LambdaBuilder
```

Comment:
Maybe a little doxygen comment on top why this builder is here and what is the difference of it compared to the LambdaNode::Create() and LambdaNode::Finalize() methods.

Reply:
This should eventually replace all other means of constructing lambdas -- this PR is for illustrating and discussing whether you are fine with the shape before I start converting anything.